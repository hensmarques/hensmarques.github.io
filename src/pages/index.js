import React from "react"

import Layout from "../components/layout"
import About from "../components/about"
import Timeline from "../components/timeline"
import Portfolio from "../components/portfolio"
import Footer from "../components/footer"

import SEO from "../components/seo"

const IndexPage = () => (
    <Layout>
        <SEO title="Home" /> 
        <About></About>
        <Timeline></Timeline>
        <div className="mb-24"></div>
        <Portfolio></Portfolio>
        <Footer></Footer>
    </Layout>
)
    
export default IndexPage
    