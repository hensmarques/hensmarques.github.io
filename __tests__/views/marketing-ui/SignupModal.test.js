import React from 'react';
import { SignupModal } from '../../../src/views/marketing-ui/SignupModal';
import { status } from '../../../src/constants';

const mockSignupData = {
  city: 'mockCity',
  country: 'mockCountry',
  dateOfBirth: '01-01-1990',
  email: 'foo@bar.com',
  givenName: 'Foo',
  middleName: 'Bar',
  password: 'password',
  phoneNumber: 123456,
  postalCode: 456789,
  stateProvince: 'mockState',
  streetAddress: 'mockStreetAddress',
  surname: 'Biz'
};

const initialProps = {
  signupStatus: status.pending,
  signupError: null,
  resendVerificationStatus: null,
  resendVerification: jest.fn(),
  signup: jest.fn(),
  onClose: jest.fn(),
  getProfileSchema: jest.fn(),
  affPrograms: {
    enabled: false,
    items: [],
  },
};
let wrapper;

describe('<SignupModal/>', () => {
  describe('Basic rendering', () => {
    it('the signup form matches the snapshot', () => {
      wrapper = shallow(<SignupModal {...initialProps} />);
      expect(wrapper).toMatchSnapshot();
    });

    it('sets formState and checks if passwords match on change', () => {
      const mockFormState = {
        'password': { value: 'password' },
        'confirm_password': { value: 'passworddd' }
      };
      wrapper = shallow(<SignupModal {...initialProps} />);
      const passwordConfirmedSpy = jest.spyOn(
        wrapper.instance(),
        'passwordConfirmed'
      );
      expect(wrapper.state().formState).toEqual({});
      expect(wrapper.state().passwordsMatch).toBe(false);

      // Simulate non-matching passwords
      wrapper.instance().onFormChange(mockFormState);
      expect(wrapper.state().formState).toEqual(mockFormState);
      expect(wrapper.state().passwordsMatch).toBe(false);
      expect(passwordConfirmedSpy).toHaveBeenCalledWith(
        mockFormState['password'].value,
        mockFormState['confirm_password'].value
      );

      // Simulate matching passwords
      mockFormState['confirm_password'].value = 'password';
      wrapper.instance().onFormChange(mockFormState);
      expect(wrapper.state().formState).toEqual(mockFormState);
      expect(wrapper.state().passwordsMatch).toBe(true);
    });
  });

  describe('Form submission', () => {
    it('is only allowed when passwords match', () => {
      wrapper = shallow(<SignupModal {...initialProps} />);
      expect(wrapper.state().passwordsMatch).toBe(false);
      
      // Dynamic rendering of the form post retrieving the schema from the backend
      // means that the following test fails because we haven't built the form container
      // right away. Should probably rework this test but commenting out for now
      
      //expect(wrapper.find('Connect(FormContainer)').props().allowSubmit).toBe(false);
      //wrapper.setState({ passwordsMatch: true });
      //expect(wrapper.find('Connect(FormContainer)').props().allowSubmit).toBe(true);
    });

    it('dispatches signup when form is submitted', () => {
      wrapper = shallow(<SignupModal {...initialProps} />);
      initialProps.signup.mockClear();
      wrapper.instance().submit(mockSignupData);
      expect(initialProps.signup.mock.calls[0]).toEqual([
        {
          password: mockSignupData.password,
          email_address: mockSignupData.email,
          custom_profile: JSON.stringify({
            city: mockSignupData.city,
            country: mockSignupData.country,
            date_of_birth:
              new Date(mockSignupData.dateOfBirth).getTime() / 1000,
            given_name: mockSignupData.givenName,
            middle_name: mockSignupData.middleName,
            phone_number: mockSignupData.phoneNumber,
            postal_code: mockSignupData.postalCode,
            state_province: mockSignupData.stateProvince,
            street_address: mockSignupData.streetAddress,
            surname: mockSignupData.surname
          })
        }
      ]);
    });
  });

  describe('Signup success', () => {
    const successProps = Object.assign({}, initialProps, {
      signupStatus: status.success
    });

    beforeEach(() => {
      wrapper = shallow(<SignupModal {...successProps} />);
      successProps.resendVerification.mockClear();
    });

    it('renders a success message', () => {
      expect(
        wrapper.containsMatchingElement(<h3>Registration Almost Complete</h3>)
      ).toBe(true);
    });

    it('dispatches resendVerification when the resend verification email link is clicked', () => {
      const mockEmail = 'foo@bar.com';
      wrapper.setState({ signupData: { email: mockEmail } });
      wrapper.find('.link').simulate('click');
      expect(successProps.resendVerification).toHaveBeenCalledWith(mockEmail);
    });
  });

  describe('Signup error', () => {
    const errorProps = Object.assign({}, initialProps, {
      signupStatus: status.failed,
      signupError: {
        error: 'DUPLICATED_USERNAME',
        error_description: 'Some description.'
      }
    });

    it('renders a duplicated username error message', () => {
      wrapper = shallow(<SignupModal {...errorProps} />);
      expect(
        wrapper.containsMatchingElement(
          <p>
            An account with that email has already been created. Check your
            inbox for a verification email.
          </p>
        )
      ).toBe(true);
    });

    it('renders a resent verification email message', () => {
      errorProps.resendVerificationStatus = 'accepted';
      wrapper = shallow(<SignupModal {...errorProps} />);
      expect(
        wrapper.find('Connect(Modal)').props().children.props.children
      ).toBe(
        'Your verification email has been resent. Please check your inbox.'
      );
    });

    it('renders a custom error message', () => {
      const translationSpy = jest.spyOn(window, '_t').mockClear();
      errorProps.signupStatus = status.failed;
      errorProps.signupError.error_description = 'Mock error message';
      errorProps.signupError.error = 'MOCK_STATUS_CODE';
      wrapper = shallow(<SignupModal {...errorProps} />);
      expect(
        wrapper.find('Connect(Modal)').props().children.props.children
      ).toBe(errorProps.signupError.error_description);
      expect(translationSpy).toHaveBeenCalledWith(
        errorProps.signupError.error_description,
        `MARKETING.SIGNUP_ERROR_${errorProps.signupError.error}`
      );
    });

    it('renders a generic error message', () => {
      errorProps.resendVerificationStatus = null;
      errorProps.signupStatus = status.failed;
      errorProps.signupError = {};
      wrapper = shallow(<SignupModal {...errorProps} />);
      expect(
        wrapper.find('Connect(Modal)').props().children.props.children
      ).toBe('There was an error registering your account, please try again');
    });
  });
});
