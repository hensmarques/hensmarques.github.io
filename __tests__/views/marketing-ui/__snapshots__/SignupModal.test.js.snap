// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignupModal/> Basic rendering the signup form matches the snapshot 1`] = `
<div>
  <Connect(Modal)
    allowSubmit={false}
    className="marketing-modal"
  >
    <Spinner />
  </Connect(Modal)>
  <Link
    replace={false}
    to="/home"
  >
    <img
      className="xIcon"
      src="images/icons/xThin.svg"
    />
  </Link>
</div>
`;
